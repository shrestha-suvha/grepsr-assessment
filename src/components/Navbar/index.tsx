import React, { FC } from 'react';
import { Col, Row } from 'react-bootstrap';
import { FiSearch } from 'react-icons/fi';
import { BiChevronLeft } from 'react-icons/bi';

import { NavbarProps } from './types';

const Navbar: FC<NavbarProps> = ({ title }) => {
  return (
    <Row
      className="top-nav ps-4 pe-4 d-flex justify-content-start align-items-center border-bottom"
      data-testid="navbar-1"
    >
      <Col className="ps-0 ">
        <div className="rounded-2 d-inline-flex  justify-content-center  align-items-center top-nav__back-arrow">
          <BiChevronLeft size="1em" color="#1276ff" />
        </div>
        <span className="top-nav__title fw-bold fs-4 line-height-lg ps-2">{title}</span>
      </Col>
      <Col className="">
        <div className="input-group  input-group-sm">
          <span className="input-group-text bg-transparent">
            <FiSearch size="1.5em" color="gray" />
          </span>
          <input
            type="text"
            className="form-control border-start-0 top-nav__search"
            placeholder="Search datasets"
            aria-label="Search datasets"
          />
        </div>
      </Col>
    </Row>
  );
};
export default Navbar;
