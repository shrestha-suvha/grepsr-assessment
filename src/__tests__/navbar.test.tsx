import { render, screen, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

import Navbar from 'src/components/Navbar';

afterEach(() => {
  cleanup();
});

test('should render navbar component', () => {
  render(<Navbar title={''} />);
  const navbarElement = screen.getByTestId('navbar-1');
  const searchFieldElement = screen.queryByPlaceholderText('Search datasets');
  expect(navbarElement).toBeInTheDocument();
  expect(searchFieldElement).toBeInTheDocument();
});

test('should render navbar component with props title', () => {
  render(<Navbar title={'Test title'} />);
  const navbarElement = screen.getByTestId('navbar-1');
  expect(navbarElement).toHaveTextContent('Test title');
});

test('matches snapshot', () => {
  const tree = create(<Navbar title={'title'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
