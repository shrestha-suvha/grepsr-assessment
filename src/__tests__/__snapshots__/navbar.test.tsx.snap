// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="top-nav ps-4 pe-4 d-flex justify-content-start align-items-center border-bottom row"
  data-testid="navbar-1"
>
  <div
    className="ps-0  col"
  >
    <div
      className="rounded-2 d-inline-flex  justify-content-center  align-items-center top-nav__back-arrow"
    >
      <svg
        color="#1276ff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#1276ff",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.293 6.293 7.586 12l5.707 5.707 1.414-1.414L10.414 12l4.293-4.293z"
        />
      </svg>
    </div>
    <span
      className="top-nav__title fw-bold fs-4 line-height-lg ps-2"
    >
      title
    </span>
  </div>
  <div
    className="col"
  >
    <div
      className="input-group  input-group-sm"
    >
      <span
        className="input-group-text bg-transparent"
      >
        <svg
          color="gray"
          fill="none"
          height="1.5em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": "gray",
            }
          }
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
      </span>
      <input
        aria-label="Search datasets"
        className="form-control border-start-0 top-nav__search"
        placeholder="Search datasets"
        type="text"
      />
    </div>
  </div>
</div>
`;
